thespian/system/transport/IPBase.py

```diff
@@ -114,13 +113,16 @@ def __init__(self, af, socktype, proto, baseaddr, port, external=False):
             baseaddr = None
         if baseaddr == '':
             baseaddr = None if external else '127.0.0.1'
-        base2 = os.getenv('THESPIAN_BASE_IPADDR', None)
-        if base2:
+        if baseaddr is None and (base2 := os.getenv('THESPIAN_BASE_IPADDR', None)):
```

Comment:
Using the walrus operator requires Python 3.8 or newer.  I'd like to be able to support earlier versions than 3.8 due to OS version limitations for some environments.  Would you mind changing this to use conventional operators?

Reply:
Sure, see https://github.com/thespianpy/Thespian/pull/87/commits/b38ddcf998072150b39879c4d94fda6ca00ca821.